### ISSUE TITLE
Tuple deconstruction Systax triggers a false positive S3459

### ISSUE THREAD

**Issue by wertzui** [SonarSource/sonar-csharp#1153]
### Description

When using the tuple deconstruction Syntax to set a private readonly variable in a constructor, S3459 will show a warning.
Maybe this happens for non-readonly members too, I have not tested this.

### Repro steps

```
public class MyClass
{
    private readonly int _foo;
    private readonly int _bar;

    public MyClass()
    {
        (_foo, _bar) = GetFoobar();
    }

    private static (int f, int b) GetFoobar() => (1, 2);

    public void DoSomething() => Console.WriteLine($"({_foo}, {_bar})");
}
```

### Expected behavior

No warning is triggered.

### Actual behavior

I see warnings telling me that `_foo` and `_bar` are not set.

### Known workarounds

None

### Related information

* SonarAnalyzer.CSharp 6.7.1.4347 NuGet Package
* Visual Studio 2017

**Comment by Evangelink** [comment]
Hi @wertzui, 

We have indeed a problem with the tuple syntax because we still rely on Roslyn 1.0 we are not receiving a correct syntax tree and some of our rules are then not behaving properly.

This is a big problem and we are aware of it but we haven't been able (yet) to find a convenient solution to support newer versions of Roslyn while maintaining compatibility with older.

I am afraid we won't be able to do much for the time being.
